### PR TITLE
Added bundle registry

### PR DESCRIPTION
**Pull request by 3rfan**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Related issues/PRs | https://github.com/sulu/sulu-minimal/pull/63
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

The BundleRegistry waits for all the bundles to register.

#### Why?

We have JS files in the different bundles.

#### Example Usage

~~~js
import {bundleReady, bundlesReadyPromise} from './services/Bundles';

// In AdminBundle
Promise.all([
    ...,
    ...,
    bundlesReadyPromise(),
]).then(startApplication);

// In Bundle
bundleReady();
~~~

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/index.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ import Application from './containers/Application';
 import {viewStore} from './containers/ViewRenderer';
 import Form from './views/Form';
 import List from './views/List';
+import BundleRegistry from './BundleRegistry';
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/js/index.js]: `bundleRegistry`

File: src/Sulu/Bundle/AdminBundle/Resources/js/BundleRegistry.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// @flow
+
+const TIMEOUT = 2000;
+
+class BundleRegistry {
+    registeredBundlesCount: number = 0;
+
+    resolve: () => void;
+
+    add() {
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/js/BundleRegistry.js]: `register` ???

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/Bundles/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+The job of the Bundles service is to offer methods to guarantee that all index JavaScript files of every bundle
+is already loaded before the application gets initialized.
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/services/Bundles/README.md]: `are already`

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/Bundles/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+Promise.all([
+    ...,
+    ...,
+    bundlesReadyPromise,
+]).then(startApplication);
+
+bundleReady();
+```
+
+In another bundle we just have to call `bundleReady`.
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/services/Bundles/README.md]: `... at the very end of the index file`.

**Pull request merged**